test/concerns/sort_by_ancestry_test.rb

```diff
@@ -8,40 +8,56 @@ def test_sort_by_ancestry
       #     - n3
       #     - n4
       #   - n5
+      #     - n6
       n1 = model.create!
       n2 = model.create!(:parent => n1)
       n3 = model.create!(:parent => n2)
       n4 = model.create!(:parent => n2)
       n5 = model.create!(:parent => n1)
+      n6 = model.create!(:parent => n5)
+      nodes = [n1, n2, n3, n4, n5, n6]
 
-      records = model.sort_by_ancestry(model.all.sort_by(&:id).reverse)
-      if records[1] == n2
-        if records[2] == n3
-          assert_equal [n1, n2, n3, n4, n5].map(&:id), records.map(&:id)
-        else
-          assert_equal [n1, n2, n4, n3, n5].map(&:id), records.map(&:id)
-        end
-      else
-        if records[3] == n3
-          assert_equal [n1, n5, n2, n3, n4].map(&:id), records.map(&:id)
-        else
-          assert_equal [n1, n5, n2, n4, n3].map(&:id), records.map(&:id)
-        end
-      end
+      # n1 needs to move to front, and n2 needs to move in front of n4, n3
+      assert_equal [n1, n5, n6, n2, n4, n3].map(&:id), model.sort_by_ancestry(model.all.sort_by(&:id).reverse).map(&:id)
+      # none are parents
+      #assert_equal [n5, n4, n3].map(&:id), model.sort_by_ancestry([n5, n4, n3]).map(&:id)
```

Comment:
Does this not work?

Reply:
My definition of proper ordering is:

```
model.sort_by_ancestry(model.all).select {|n| node_of_concern(n)} ===
  model.sort_by_ancestry(model.all.select {|n| node_of_concern(n)})
```

In these tests our ordering is:
```
      # - n1 (0)
      #   - n5 (0)
      #   - n3 (3)
      # - n2 (1)
      #   - n4 (0)
      #   - n6 (1)
```

The `n1 < n2` need to be present to properly place their children `(n5, n3) < (n4, n6)`
The children have this order not because of any of their own information, but rather because of their parent's information.

I feel I've beaten this horse to death in this round of comments. Please let me know if there is more I can provide to make more clear.